--- compile-abstract/MAlonzo/Code/Data/Fin/VecCat/PullbackB.hs
+++ compile-float/MAlonzo/Code/Data/Fin/VecCat/PullbackB.hs
@@ -98,15 +98,17 @@
 name60 = "Data.Fin.VecCat.PullbackB.FinPB₀._.s"
 d60 v0 v1 v2 v3 v4 v5 v6 v7 = du60 v1 v2 v3 v4 v5
 du60 v0 v1 v2 v3 v4
-  = let v5 = coe du58 v0 v1 v2 v3 v4 in
-    case coe v5 of
+  = let v5 = coe du58 v0 v1 v2 v3 v4 in dv60 v0 v1 v2 v3 v4 v5
+dv60 v0 v1 v2 v3 v4 v5
+  = case coe v5 of
       MAlonzo.Code.Data.Product.C30 v6 v7 -> coe v6
       _ -> coe MAlonzo.RTE.mazUnreachableError
 name62 = "Data.Fin.VecCat.PullbackB.FinPB₀._.n₀"
 d62 v0 v1 v2 v3 v4 v5 v6 v7 = du62 v1 v2 v3 v4 v5
 du62 v0 v1 v2 v3 v4
-  = let v5 = coe du58 v0 v1 v2 v3 v4 in
-    case coe v5 of
+  = let v5 = coe du58 v0 v1 v2 v3 v4 in dv62 v0 v1 v2 v3 v4 v5
+dv62 v0 v1 v2 v3 v4 v5
+  = case coe v5 of
       MAlonzo.Code.Data.Product.C30 v6 v7
         -> case coe v7 of
              MAlonzo.Code.Data.Product.C30 v8 v9 -> coe v8
@@ -115,8 +117,9 @@
 name70 = "Data.Fin.VecCat.PullbackB.FinPB₀._.v"
 d70 v0 v1 v2 v3 v4 v5 v6 v7 = du70 v1 v2 v3 v4 v5
 du70 v0 v1 v2 v3 v4
-  = let v5 = coe du58 v0 v1 v2 v3 v4 in
-    case coe v5 of
+  = let v5 = coe du58 v0 v1 v2 v3 v4 in dv70 v0 v1 v2 v3 v4 v5
+dv70 v0 v1 v2 v3 v4 v5
+  = case coe v5 of
       MAlonzo.Code.Data.Product.C30 v6 v7
         -> case coe v7 of
              MAlonzo.Code.Data.Product.C30 v8 v9 -> coe v9
