--- compile/MAlonzo/Code/GraTra/Graphs.hs
+++ compile-inline-genplet-float/MAlonzo/Code/GraTra/Graphs.hs
@@ -11574,64 +15068,58 @@
-name4788 = "GraTra.Graphs.Coproducts-Graph.coproduct"
-d4788 v0 v1
-  = coe
-      MAlonzo.Code.Categoric.FinColimits.Coproduct.C8807
-      (coe
-         MAlonzo.Code.Categoric.FinColimits.Coproduct.d1062
-         (coe
-            MAlonzo.Code.Data.Algebra.Generalised.Hom.Coproduct.du1112
-            MAlonzo.Code.Data.Fin.VecCat.d26 MAlonzo.Code.Data.Fin.VecCat.d3920
-            MAlonzo.Code.Data.Algebra.Signature.Graph.d20)
-         v0 v1)
-      (coe
-         MAlonzo.Code.Categoric.FinColimits.Coproduct.d1068
-         (coe
-            MAlonzo.Code.Data.Algebra.Generalised.Hom.Coproduct.du1112
-            MAlonzo.Code.Data.Fin.VecCat.d26 MAlonzo.Code.Data.Fin.VecCat.d3920
-            MAlonzo.Code.Data.Algebra.Signature.Graph.d20)
-         v0 v1)
-      (coe
-         MAlonzo.Code.Categoric.FinColimits.Coproduct.d1074
-         (coe
-            MAlonzo.Code.Data.Algebra.Generalised.Hom.Coproduct.du1112
-            MAlonzo.Code.Data.Fin.VecCat.d26 MAlonzo.Code.Data.Fin.VecCat.d3920
-            MAlonzo.Code.Data.Algebra.Signature.Graph.d20)
-         v0 v1)
-      (coe
-         MAlonzo.Code.Categoric.FinColimits.Coproduct.d1080
-         (coe
+name4788 = "GraTra.Graphs.Coproducts-Graph.coproduct"
+d4788 v0 v1
+  = let v2@(MAlonzo.Code.Categoric.FinColimits.Coproduct.C9169 v3 v4 v5 v6)
+          = coe
             MAlonzo.Code.Data.Algebra.Generalised.Hom.Coproduct.du1112
             MAlonzo.Code.Data.Fin.VecCat.d26 MAlonzo.Code.Data.Fin.VecCat.d3920
-            MAlonzo.Code.Data.Algebra.Signature.Graph.d20)
-         v0 v1)
+              MAlonzo.Code.Data.Algebra.Signature.Graph.d20 in
+    dv4788 v0 v1 v2 v3 v4 v5 v6
+dv4788 v0 v1 v2 v3 v4 v5 v6
+  = coe
+      MAlonzo.Code.Categoric.FinColimits.Coproduct.C8807 (coe v3 v0 v1)
+      (coe v4 v0 v1) (coe v5 v0 v1) (coe v6 v0 v1)
